nexus/db-queries/src/db/datastore/ip_pool.rs

```diff
@@ -65,44 +63,100 @@ impl DataStore {
                 &pagparams.map_name(|n| Name::ref_cast(n)),
             ),
         }
-        .filter(dsl::internal.eq(false))
+        // != excludes nulls so we explicitly include them
+        .filter(dsl::silo_id.ne(*INTERNAL_SILO_ID).or(dsl::silo_id.is_null()))
         .filter(dsl::time_deleted.is_null())
         .select(db::model::IpPool::as_select())
         .get_results_async(self.pool_authorized(opctx).await?)
         .await
         .map_err(|e| public_error_from_diesel_pool(e, ErrorHandler::Server))
     }
 
-    /// Looks up the default IP pool by name.
+    /// Looks up the default IP pool for a given scope, i.e., a given
+    /// combination of silo and project ID. If there is no default at a given
+    /// scope, fall back up a level. There should always be a default at fleet
+    /// level, though this query can theoretically fail.
     pub async fn ip_pools_fetch_default_for(
         &self,
         opctx: &OpContext,
-        action: authz::Action,
-    ) -> LookupResult<(authz::IpPool, IpPool)> {
-        self.ip_pools_fetch_for(
-            opctx,
-            action,
-            &Name(ExternalName::from_str("default").unwrap()),
-        )
-        .await
+        // Optional primarily because there are test contexts where we don't care
+        // about the project. If project ID is None, we will only get back pools
+        // that themselves have no project.
```

Comment:
```suggestion
        // that themselves have no associated project.
```

... does that mean we'd just get fleet-wide pools?

Reply:
Or silo-scoped pools. The signature here is a little misloading because it only takes project (because we are pulling the current silo off `opctx`) but the logic does look at both silo and project ID when picking defaults.